### DISCUSSION TITLE
rueidislock : panic: locker closed

### DISCUSSION THREAD

**Discussion by leishihao521** [redis/rueidis#240] [Q&A]
When I use rueidislock, I will wait at _, cancel, err := locker.WithContext(ctx, "my_lock11112312") and finally report: panic: locker closed. What is the reason ，I'm using cluster mode

**Comment by rueian** [comment]
Hi @leishihao521,

Could you provide a sample code for your case? Normally the error indicates that your connection to redis is broken or you have invoked `locker.Close()` somewhere.

**Reply by leishihao521** [reply]
//rueidislock
	locker, err := rueidislock.NewLocker(rueidislock.LockerOption{
		ClientOption: rueidis.ClientOption{InitAddress: []string{"127.0.0.1:7000", "127.0.0.1:7001", "127.0.0.1:7002"}, Password: "123456"},
		KeyMajority:  2,
	})
	if err != nil {
		panic(err)
	}
	defer locker.Close()

	now := time.Now()
	// acquire the lock "my_lock"
	fmt.Println("Current time", now)
	ctx, cancel, err := locker.WithContext(context.Background(), "my_lock")
	if err != nil {
 
		fmt.Println(err.Error())
		panic(err)
	}

	fmt.Println("Acquired lock", time.Since(now), ctx)
 
	defer cancel()
	fmt.Println("Release lock", time.Since(now))

// Stop at ctx, cancel, err := locker.WithContext(context.Background(), "my_lock") and report: locker closed
panic: locker closed
There is no problem in the local cluster test. This will happen in the local connection test environment. Is there any reason for this?

**Reply by leishihao521** [reply]
![image](https://github.com/redis/rueidis/assets/54799975/40e94bd2-e6d0-4ff5-bd39-ad794b19d581)
Here's how the test environment creates the lock

**Reply by rueian** [reply]
Hi, @leishihao521. sorry for the late reply. Your code looks good to me.

The error `locker closed` means that there are disconnections to your redis nodes or `FLUSH` or other events that make redis want to invalidate all your client-side caching. We may be able to add an auto-retry mechanism in future versions.

> There is no problem in the local cluster test. This will happen in the local connection test environment.

What is the difference between these two environments?
